src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -87,7 +89,17 @@ internal NpgsqlRawCopyStream(NpgsqlConnector connector, string copyCommand)
 
         #region Write
 
-        public override void Write(byte[] buffer, int offset, int count)
+        public override void Write(byte[] buffer, int offset, int count) => Write(buffer, offset, count, false).GetAwaiter().GetResult();
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return Task.FromCanceled(cancellationToken);
+            using (NoSynchronizationContextScope.Enter())
+                return Write(buffer, offset, count, true);
+        }
+
+        async Task Write(byte[] buffer, int offset, int count, bool async)
```

Comment:
This method internally calls synchronous Flush even in the async path.

The class needs to implement FlushAsync.